pandas/_libs/algos.pyx

```diff
@@ -793,216 +796,280 @@ ctypedef fused rank_t:
 @cython.boundscheck(False)
 def rank_1d(
     ndarray[rank_t, ndim=1] in_arr,
+    const int64_t[:] labels,
     ties_method="average",
     bint ascending=True,
-    na_option="keep",
     bint pct=False,
+    na_option="keep",
 ):
     """
     Fast NaN-friendly version of ``scipy.stats.rankdata``.
+
+    Parameters
+    ----------
+    in_arr : array of rank_t values to be ranked
```

Comment:
minor nit, I *think* we should be more consistent on the naming of these args we call this (in_arr, arr, values) in several different places. I think we *don't* use in_arr  except here, i would rename to arr or values.

Reply:
Changed to `values`. `rank_2d` also has `in_arr` (only other place I see it), can followup after this pr to try to unify naming / logic that can be deduplicated between `rank_2d` and `rank_1d`